Tensile/Tests/unit/test_TensileCreateLibrary.py

```diff
@@ -266,9 +266,56 @@ def verifyYamlAndYml():
     assert verifyYamlAndYml(), "Output should have twice as many files as old logic (which only parses .yaml)"
 
 
+def test_createClientConfig():
+    
+    outputPath: Path = Path.cwd() / "my-output"
+    masterLibrary: Path = outputPath / "masterlib.data" 
+    codeObjectFiles = ["library/foo.hsaco", "library/bar.co"]
+    configFile = outputPath / "best-solution.ini"
+    
+    cleanClientConfigTest(outputPath, masterLibrary, codeObjectFiles, configFile)
+
+    with pytest.raises(FileNotFoundError, match=r"(.*) No such file or directory: '(.*)/my-output/best-solution.ini'"): 
+        TensileCreateLibrary.createClientConfig(outputPath, masterLibrary, codeObjectFiles)        
+
+    setupClientConfigTest(outputPath, masterLibrary, codeObjectFiles)
+
+    TensileCreateLibrary.createClientConfig(outputPath, masterLibrary, codeObjectFiles)
+
+    assert configFile.is_file(), "{configFile} was not generated"
+    
+    with open(configFile, "r") as f:
+        result = f.readlines()
+        assert "library-file" in result[0], "missing library-file entry"
+        assert "code-object" in result[1], "missing code-object entry"
+        assert "code-object" in result[2], "missing code-object entry"        
+        assert "best-solution" in result[3], "missing best-solution entry"                
+
+    cleanClientConfigTest(outputPath, masterLibrary, codeObjectFiles, configFile)
+
 # ----------------
 # Helper functions
 # ----------------
+def setupClientConfigTest(outputPath, masterLibrary, codeObjectFiles):
+    outputPath.mkdir()
+    with open(masterLibrary, "w") as testFile:
+        testFile.write("foo")
+    (outputPath / "library").mkdir()
+    for f in codeObjectFiles:
+        with open(outputPath / f, "w") as testFile:
+            testFile.write("foo")
+
+
+def cleanClientConfigTest(outputPath: Path, masterLibrary, codeObjectFiles, configFile):
+    with contextlib.suppress(FileNotFoundError):
+        os.remove(configFile)
+        os.remove(masterLibrary)        
+        for f in codeObjectFiles:
+            os.remove(outputPath / f)
+        next(outputPath.iterdir()).rmdir()
+        outputPath.rmdir()
```

Comment:
Why do have to get the next item in the iterator, remove it, and then remove the `outputDir`? Is L315 required?

Reply:
Removing the subdir from the parent dir before removing the parent dir. Path doesn't provide a nice way of doing this from what I could tell. I was trying to avoid pulling in another dependency to get this done cleaner.

Reply:
I see, this is for the **library** directory that you create, makes sense.